src/user/digest.js

```diff
@@ -70,7 +103,7 @@ Digest.send = async function (data) {
 
 	const users = await user.getUsersFields(data.subscribers, ['uid', 'username', 'userslug', 'lastonline']);
 
-	async.eachLimit(users, 100, async function (userObj) {
+	await async.eachLimit(users, 100, async function (userObj) {
```

Comment:
I think `await` is needed here, otherwise `emailsSent` is always 0

Reply:
Yes, this should fix emailsSend being 0, https://github.com/NodeBB/NodeBB/issues/8010

Although the behaviour of `Digest.send` has changed after adding await to this and `emailer.send`. Before this function wasn't waiting for emailer.send to finish. It was starting the digest and returning immediately. Now it will wait for all emails to be sent before returning. So with a lot of users clicking executed digest on the ACP might take a long time if there are alot of users.

Reply:
Hm... I was afraid this might happen... though wouldn't it also be ok if `Digest.send` (or `Digest.execute`) was called without `await`? Or does it specifically have to be this `async.eachLimit` call...